desc: --pages-as-heap=yes
cmd: python ../../stime.py 3 ../../hello.exe
time_unit: i
#-----------
snapshot=0
#-----------
time=0
mem_heap_B=2560000
mem_heap_extra_B=0
mem_stacks_B=0
heap_tree=empty
#-----------
snapshot=1
#-----------
time=100279
mem_heap_B=30035968
mem_heap_extra_B=0
mem_stacks_B=0
heap_tree=empty
#-----------
snapshot=2
#-----------
time=48657766
mem_heap_B=165990400
mem_heap_extra_B=0
mem_stacks_B=0
heap_tree=peak
n4: 165990400 (page allocation syscalls) mmap/mremap/brk, --alloc-fns, etc.
 n3: 111468544 0x5E70169: mmap (syscall-template.S:82)
  n1: 83079168 0x5DFE377: new_heap (arena.c:554)
   n1: 83079168 0x5DFF2B3: arena_get2.isra.5.part.6 (arena.c:712)
    n1: 83079168 0x5E031E7: malloc (arena.c:824)
     n1: 83079168 0x4D3286: ??? (in /usr/bin/python2.7)
      n1: 83079168 0x4D348A: PyGILState_Ensure (in /usr/bin/python2.7)
       n1: 83079168 0x64E2DB9: SandboxPolicy_entry (in /usr/local/lib/python2.7/dist-packages/sandbox/_sandbox.so)
        n1: 83079168 0x66EDE5E: sandbox_watcher (in /usr/lib/libsandbox.so.0.3.5)
         n0: 83079168 0x4C2EE98: start_thread (pthread_create.c:308)
  n3: 25178112 0x4C2FA2C: pthread_create@@GLIBC_2.2.5 (allocatestack.c:495)
   n1: 8392704 0x400F304: call_init.part.0 (dl-init.c:85)
    n1: 8392704 0x400F3DD: _dl_init (dl-init.c:52)
     n1: 8392704 0x4013AD8: dl_open_worker (dl-open.c:549)
      n1: 8392704 0x400F174: _dl_catch_error (dl-error.c:178)
       n1: 8392704 0x4013318: _dl_open (dl-open.c:639)
        n1: 8392704 0x4E44F24: dlopen_doit (dlopen.c:67)
         n1: 8392704 0x400F174: _dl_catch_error (dl-error.c:178)
          n1: 8392704 0x4E4552D: _dlerror_run (dlerror.c:164)
           n1: 8392704 0x4E44FBF: dlopen@@GLIBC_2.2.5 (dlopen.c:88)
            n1: 8392704 0x561838: _PyImport_GetDynLoadFunc (in /usr/bin/python2.7)
             n1: 8392704 0x47AF17: _PyImport_LoadDynamicModule (in /usr/bin/python2.7)
              n1: 8392704 0x53D7D9: ??? (in /usr/bin/python2.7)
               n1: 8392704 0x53DCA8: ??? (in /usr/bin/python2.7)
                n1: 8392704 0x53E1B1: ??? (in /usr/bin/python2.7)
                 n1: 8392704 0x4C2E69: ??? (in /usr/bin/python2.7)
                  n1: 8392704 0x4E9F34: PyObject_Call (in /usr/bin/python2.7)
                   n1: 8392704 0x4EA394: PyEval_CallObjectWithKeywords (in /usr/bin/python2.7)
                    n1: 8392704 0x49A4EB: PyEval_EvalFrameEx (in /usr/bin/python2.7)
                     n1: 8392704 0x49F1BE: PyEval_EvalCodeEx (in /usr/bin/python2.7)
                      n1: 8392704 0x4A7F16: PyImport_ExecCodeModuleEx (in /usr/bin/python2.7)
                       n1: 8392704 0x53CDDF: ??? (in /usr/bin/python2.7)
                        n1: 8392704 0x4A4189: ??? (in /usr/bin/python2.7)
                         n1: 8392704 0x53D7D9: ??? (in /usr/bin/python2.7)
                          n1: 8392704 0x4FC1A1: ??? (in /usr/bin/python2.7)
                           n1: 8392704 0x53E0A2: ??? (in /usr/bin/python2.7)
                            n1: 8392704 0x4C2E69: ??? (in /usr/bin/python2.7)
                             n1: 8392704 0x4E9F34: PyObject_Call (in /usr/bin/python2.7)
                              n0: 8392704 0x4EA394: PyEval_CallObjectWithKeywords (in /usr/bin/python2.7)
   n1: 8392704 0x66F02FD: sandbox_execute (in /usr/lib/libsandbox.so.0.3.5)
    n1: 8392704 0x64E1D7C: Sandbox_run (in /usr/local/lib/python2.7/dist-packages/sandbox/_sandbox.so)
     n1: 8392704 0x498908: PyEval_EvalFrameEx (in /usr/bin/python2.7)
      n1: 8392704 0x498600: PyEval_EvalFrameEx (in /usr/bin/python2.7)
       n1: 8392704 0x49F1BE: PyEval_EvalCodeEx (in /usr/bin/python2.7)
        n1: 8392704 0x4A907F: PyRun_FileExFlags (in /usr/bin/python2.7)
         n1: 8392704 0x4A930F: PyRun_SimpleFileExFlags (in /usr/bin/python2.7)
          n1: 8392704 0x4AA8BB: Py_Main (in /usr/bin/python2.7)
           n0: 8392704 0x5DA176B: (below main) (libc-start.c:226)
   n1: 8392704 0x66F0322: sandbox_execute (in /usr/lib/libsandbox.so.0.3.5)
    n1: 8392704 0x64E1D7C: Sandbox_run (in /usr/local/lib/python2.7/dist-packages/sandbox/_sandbox.so)
     n1: 8392704 0x498908: PyEval_EvalFrameEx (in /usr/bin/python2.7)
      n1: 8392704 0x498600: PyEval_EvalFrameEx (in /usr/bin/python2.7)
       n1: 8392704 0x49F1BE: PyEval_EvalCodeEx (in /usr/bin/python2.7)
        n1: 8392704 0x4A907F: PyRun_FileExFlags (in /usr/bin/python2.7)
         n1: 8392704 0x4A930F: PyRun_SimpleFileExFlags (in /usr/bin/python2.7)
          n1: 8392704 0x4AA8BB: Py_Main (in /usr/bin/python2.7)
           n0: 8392704 0x5DA176B: (below main) (libc-start.c:226)
  n0: 3211264 in 4 places, all below massif's threshold (01.00%)
 n3: 50274304 0x4018429: mmap (syscall-template.S:82)
  n1: 45649920 0x4006B67: _dl_map_object_from_fd (dl-load.c:1283)
   n3: 45649920 0x4008698: _dl_map_object (dl-load.c:2568)
    n1: 37122048 0x400CFE0: openaux (dl-deps.c:65)
     n1: 37122048 0x400F174: _dl_catch_error (dl-error.c:178)
      n2: 37122048 0x400D6BC: _dl_map_object_deps (dl-deps.c:258)
       n1: 24305664 0x400350B: dl_main (rtld.c:1826)
        n1: 24305664 0x4015B22: _dl_sysdep_start (dl-sysdep.c:244)
         n1: 24305664 0x4005363: _dl_start (rtld.c:338)
          n1: 24305664 0x40016B6: ??? (in /lib/x86_64-linux-gnu/ld-2.15.so)
           n1: 24305664 0x2: ???
            n1: 24305664 0x7FF0007A5: ???
             n1: 24305664 0x7FF0007AC: ???
              n1: 24305664 0x7FF0007BB: ???
               n0: 24305664 0x7FF0007BD: ???
       n1: 12816384 0x40138A7: dl_open_worker (dl-open.c:262)
        n1: 12816384 0x400F174: _dl_catch_error (dl-error.c:178)
         n2: 12816384 0x4013318: _dl_open (dl-open.c:639)
          n1: 8462336 0x4E44F24: dlopen_doit (dlopen.c:67)
           n1: 8462336 0x400F174: _dl_catch_error (dl-error.c:178)
            n1: 8462336 0x4E4552D: _dlerror_run (dlerror.c:164)
             n1: 8462336 0x4E44FBF: dlopen@@GLIBC_2.2.5 (dlopen.c:88)
              n1: 8462336 0x561838: _PyImport_GetDynLoadFunc (in /usr/bin/python2.7)
               n1: 8462336 0x47AF17: _PyImport_LoadDynamicModule (in /usr/bin/python2.7)
                n1: 8462336 0x53D7D9: ??? (in /usr/bin/python2.7)
                 n1: 8462336 0x53DCA8: ??? (in /usr/bin/python2.7)
                  n1: 8462336 0x53E1B1: ??? (in /usr/bin/python2.7)
                   n1: 8462336 0x4C2E69: ??? (in /usr/bin/python2.7)
                    n1: 8462336 0x4E9F34: PyObject_Call (in /usr/bin/python2.7)
                     n1: 8462336 0x4EA394: PyEval_CallObjectWithKeywords (in /usr/bin/python2.7)
                      n1: 8462336 0x49A4EB: PyEval_EvalFrameEx (in /usr/bin/python2.7)
                       n1: 8462336 0x49F1BE: PyEval_EvalCodeEx (in /usr/bin/python2.7)
                        n1: 8462336 0x4A7F16: PyImport_ExecCodeModuleEx (in /usr/bin/python2.7)
                         n1: 8462336 0x53CDDF: ??? (in /usr/bin/python2.7)
                          n1: 8462336 0x4A4189: ??? (in /usr/bin/python2.7)
                           n1: 8462336 0x53D7D9: ??? (in /usr/bin/python2.7)
                            n1: 8462336 0x4FC1A1: ??? (in /usr/bin/python2.7)
                             n1: 8462336 0x53E0A2: ??? (in /usr/bin/python2.7)
                              n0: 8462336 0x4C2E69: ??? (in /usr/bin/python2.7)
          n1: 4354048 0x5EB0700: do_dlopen (dl-libc.c:89)
           n1: 4354048 0x400F174: _dl_catch_error (dl-error.c:178)
            n1: 4354048 0x5EB07C2: __libc_dlopen_mode (dl-libc.c:48)
             n1: 4354048 0x5E85B1C: nss_load_library.isra.1 (nsswitch.c:372)
              n2: 4354048 0x5E8657B: __nss_lookup_function (nsswitch.c:474)
               n1: 2203648 0x5E8678A: __nss_lookup (nsswitch.c:202)
                n1: 2203648 0x5E3EE16: getpwuid_r@@GLIBC_2.2.5 (getXXbyYY_r.c:203)
                 n1: 2203648 0x5E3E5F1: getpwuid (getXXbyYY.c:117)
                  n1: 2203648 0x66EFD6F: sandbox_check (in /usr/lib/libsandbox.so.0.3.5)
                   n1: 2203648 0x64E2D1E: Sandbox_new (in /usr/local/lib/python2.7/dist-packages/sandbox/_sandbox.so)
                    n1: 2203648 0x4DB5E6: ??? (in /usr/bin/python2.7)
                     n1: 2203648 0x4E9F34: PyObject_Call (in /usr/bin/python2.7)
                      n1: 2203648 0x498468: PyEval_EvalFrameEx (in /usr/bin/python2.7)
                       n1: 2203648 0x49F1BE: PyEval_EvalCodeEx (in /usr/bin/python2.7)
                        n1: 2203648 0x4A907F: PyRun_FileExFlags (in /usr/bin/python2.7)
                         n1: 2203648 0x4A930F: PyRun_SimpleFileExFlags (in /usr/bin/python2.7)
                          n1: 2203648 0x4AA8BB: Py_Main (in /usr/bin/python2.7)
                           n0: 2203648 0x5DA176B: (below main) (libc-start.c:226)
               n1: 2150400 0x770063D: ???
                n1: 2150400 0x5E3ECFB: getpwuid_r@@GLIBC_2.2.5 (getXXbyYY_r.c:256)
                 n1: 2150400 0x5E3E5F1: getpwuid (getXXbyYY.c:117)
                  n1: 2150400 0x66EFD6F: sandbox_check (in /usr/lib/libsandbox.so.0.3.5)
                   n1: 2150400 0x64E2D1E: Sandbox_new (in /usr/local/lib/python2.7/dist-packages/sandbox/_sandbox.so)
                    n1: 2150400 0x4DB5E6: ??? (in /usr/bin/python2.7)
                     n1: 2150400 0x4E9F34: PyObject_Call (in /usr/bin/python2.7)
                      n1: 2150400 0x498468: PyEval_EvalFrameEx (in /usr/bin/python2.7)
                       n1: 2150400 0x49F1BE: PyEval_EvalCodeEx (in /usr/bin/python2.7)
                        n1: 2150400 0x4A907F: PyRun_FileExFlags (in /usr/bin/python2.7)
                         n1: 2150400 0x4A930F: PyRun_SimpleFileExFlags (in /usr/bin/python2.7)
                          n1: 2150400 0x4AA8BB: Py_Main (in /usr/bin/python2.7)
                           n0: 2150400 0x5DA176B: (below main) (libc-start.c:226)
    n1: 6422528 0x4013849: dl_open_worker (dl-open.c:225)
     n1: 6422528 0x400F174: _dl_catch_error (dl-error.c:178)
      n2: 6422528 0x4013318: _dl_open (dl-open.c:639)
       n1: 4280320 0x5EB0700: do_dlopen (dl-libc.c:89)
        n1: 4280320 0x400F174: _dl_catch_error (dl-error.c:178)
         n1: 4280320 0x5EB07C2: __libc_dlopen_mode (dl-libc.c:48)
          n1: 4280320 0x5E85B1C: nss_load_library.isra.1 (nsswitch.c:372)
           n2: 4280320 0x5E8657B: __nss_lookup_function (nsswitch.c:474)
            n1: 2146304 0x770063D: ???
             n1: 2146304 0x5E3ECFB: getpwuid_r@@GLIBC_2.2.5 (getXXbyYY_r.c:256)
              n1: 2146304 0x5E3E5F1: getpwuid (getXXbyYY.c:117)
               n1: 2146304 0x66EFD6F: sandbox_check (in /usr/lib/libsandbox.so.0.3.5)
                n1: 2146304 0x64E2D1E: Sandbox_new (in /usr/local/lib/python2.7/dist-packages/sandbox/_sandbox.so)
                 n1: 2146304 0x4DB5E6: ??? (in /usr/bin/python2.7)
                  n1: 2146304 0x4E9F34: PyObject_Call (in /usr/bin/python2.7)
                   n1: 2146304 0x498468: PyEval_EvalFrameEx (in /usr/bin/python2.7)
                    n1: 2146304 0x49F1BE: PyEval_EvalCodeEx (in /usr/bin/python2.7)
                     n1: 2146304 0x4A907F: PyRun_FileExFlags (in /usr/bin/python2.7)
                      n1: 2146304 0x4A930F: PyRun_SimpleFileExFlags (in /usr/bin/python2.7)
                       n1: 2146304 0x4AA8BB: Py_Main (in /usr/bin/python2.7)
                        n0: 2146304 0x5DA176B: (below main) (libc-start.c:226)
            n1: 2134016 0x5E8678A: __nss_lookup (nsswitch.c:202)
             n1: 2134016 0x5E3EE16: getpwuid_r@@GLIBC_2.2.5 (getXXbyYY_r.c:203)
              n1: 2134016 0x5E3E5F1: getpwuid (getXXbyYY.c:117)
               n1: 2134016 0x66EFD6F: sandbox_check (in /usr/lib/libsandbox.so.0.3.5)
                n1: 2134016 0x64E2D1E: Sandbox_new (in /usr/local/lib/python2.7/dist-packages/sandbox/_sandbox.so)
                 n1: 2134016 0x4DB5E6: ??? (in /usr/bin/python2.7)
                  n1: 2134016 0x4E9F34: PyObject_Call (in /usr/bin/python2.7)
                   n1: 2134016 0x498468: PyEval_EvalFrameEx (in /usr/bin/python2.7)
                    n1: 2134016 0x49F1BE: PyEval_EvalCodeEx (in /usr/bin/python2.7)
                     n1: 2134016 0x4A907F: PyRun_FileExFlags (in /usr/bin/python2.7)
                      n1: 2134016 0x4A930F: PyRun_SimpleFileExFlags (in /usr/bin/python2.7)
                       n1: 2134016 0x4AA8BB: Py_Main (in /usr/bin/python2.7)
                        n0: 2134016 0x5DA176B: (below main) (libc-start.c:226)
       n1: 2142208 0x4E44F24: dlopen_doit (dlopen.c:67)
        n1: 2142208 0x400F174: _dl_catch_error (dl-error.c:178)
         n1: 2142208 0x4E4552D: _dlerror_run (dlerror.c:164)
          n1: 2142208 0x4E44FBF: dlopen@@GLIBC_2.2.5 (dlopen.c:88)
           n1: 2142208 0x561838: _PyImport_GetDynLoadFunc (in /usr/bin/python2.7)
            n1: 2142208 0x47AF17: _PyImport_LoadDynamicModule (in /usr/bin/python2.7)
             n1: 2142208 0x53D7D9: ??? (in /usr/bin/python2.7)
              n1: 2142208 0x53DCA8: ??? (in /usr/bin/python2.7)
               n1: 2142208 0x53E1B1: ??? (in /usr/bin/python2.7)
                n1: 2142208 0x4C2E69: ??? (in /usr/bin/python2.7)
                 n1: 2142208 0x4E9F34: PyObject_Call (in /usr/bin/python2.7)
                  n1: 2142208 0x4EA394: PyEval_CallObjectWithKeywords (in /usr/bin/python2.7)
                   n1: 2142208 0x49A4EB: PyEval_EvalFrameEx (in /usr/bin/python2.7)
                    n1: 2142208 0x49F1BE: PyEval_EvalCodeEx (in /usr/bin/python2.7)
                     n1: 2142208 0x4A7F16: PyImport_ExecCodeModuleEx (in /usr/bin/python2.7)
                      n1: 2142208 0x53CDDF: ??? (in /usr/bin/python2.7)
                       n1: 2142208 0x4A4189: ??? (in /usr/bin/python2.7)
                        n1: 2142208 0x53D7D9: ??? (in /usr/bin/python2.7)
                         n1: 2142208 0x4FC1A1: ??? (in /usr/bin/python2.7)
                          n1: 2142208 0x53E0A2: ??? (in /usr/bin/python2.7)
                           n1: 2142208 0x4C2E69: ??? (in /usr/bin/python2.7)
                            n1: 2142208 0x4E9F34: PyObject_Call (in /usr/bin/python2.7)
                             n1: 2142208 0x4EA394: PyEval_CallObjectWithKeywords (in /usr/bin/python2.7)
                              n0: 2142208 0x49A4EB: PyEval_EvalFrameEx (in /usr/bin/python2.7)
    n1: 2105344 0x400188C: map_doit (rtld.c:633)
     n1: 2105344 0x400F174: _dl_catch_error (dl-error.c:178)
      n1: 2105344 0x4000ECD: do_preload (rtld.c:822)
       n1: 2105344 0x4004118: dl_main (rtld.c:1717)
        n1: 2105344 0x4015B22: _dl_sysdep_start (dl-sysdep.c:244)
         n1: 2105344 0x4005363: _dl_start (rtld.c:338)
          n1: 2105344 0x40016B6: ??? (in /lib/x86_64-linux-gnu/ld-2.15.so)
           n1: 2105344 0x2: ???
            n1: 2105344 0x7FF0007A5: ???
             n1: 2105344 0x7FF0007AC: ???
              n1: 2105344 0x7FF0007BB: ???
               n0: 2105344 0x7FF0007BD: ???
  n1: 4259840 0x4007148: _dl_map_object_from_fd (dl-load.c:1392)
   n1: 4259840 0x4008698: _dl_map_object (dl-load.c:2568)
    n1: 4259840 0x400CFE0: openaux (dl-deps.c:65)
     n1: 4259840 0x400F174: _dl_catch_error (dl-error.c:178)
      n2: 4259840 0x400D6BC: _dl_map_object_deps (dl-deps.c:258)
       n1: 4202496 0x40138A7: dl_open_worker (dl-open.c:262)
        n1: 4202496 0x400F174: _dl_catch_error (dl-error.c:178)
         n2: 4202496 0x4013318: _dl_open (dl-open.c:639)
          n1: 4194304 0x4E44F24: dlopen_doit (dlopen.c:67)
           n1: 4194304 0x400F174: _dl_catch_error (dl-error.c:178)
            n1: 4194304 0x4E4552D: _dlerror_run (dlerror.c:164)
             n1: 4194304 0x4E44FBF: dlopen@@GLIBC_2.2.5 (dlopen.c:88)
              n1: 4194304 0x561838: _PyImport_GetDynLoadFunc (in /usr/bin/python2.7)
               n1: 4194304 0x47AF17: _PyImport_LoadDynamicModule (in /usr/bin/python2.7)
                n1: 4194304 0x53D7D9: ??? (in /usr/bin/python2.7)
                 n1: 4194304 0x53DCA8: ??? (in /usr/bin/python2.7)
                  n1: 4194304 0x53E1B1: ??? (in /usr/bin/python2.7)
                   n1: 4194304 0x4C2E69: ??? (in /usr/bin/python2.7)
                    n1: 4194304 0x4E9F34: PyObject_Call (in /usr/bin/python2.7)
                     n1: 4194304 0x4EA394: PyEval_CallObjectWithKeywords (in /usr/bin/python2.7)
                      n1: 4194304 0x49A4EB: PyEval_EvalFrameEx (in /usr/bin/python2.7)
                       n1: 4194304 0x49F1BE: PyEval_EvalCodeEx (in /usr/bin/python2.7)
                        n1: 4194304 0x4A7F16: PyImport_ExecCodeModuleEx (in /usr/bin/python2.7)
                         n1: 4194304 0x53CDDF: ??? (in /usr/bin/python2.7)
                          n1: 4194304 0x4A4189: ??? (in /usr/bin/python2.7)
                           n1: 4194304 0x53D7D9: ??? (in /usr/bin/python2.7)
                            n1: 4194304 0x4FC1A1: ??? (in /usr/bin/python2.7)
                             n1: 4194304 0x53E0A2: ??? (in /usr/bin/python2.7)
                              n0: 4194304 0x4C2E69: ??? (in /usr/bin/python2.7)
          n0: 8192 in 1 place, below massif's threshold (01.00%)
       n0: 57344 in 1 place, below massif's threshold (01.00%)
  n0: 364544 in 3 places, all below massif's threshold (01.00%)
 n0: 3239936 0xFFFFFFFFFFFFFFFF: ???
 n0: 1007616 in 1 place, below massif's threshold (01.00%)
#-----------
snapshot=3
#-----------
time=48657751
mem_heap_B=217128960
mem_heap_extra_B=0
mem_stacks_B=0
heap_tree=empty

